src/bucket/BucketManagerImpl.cpp

```diff
@@ -107,7 +107,9 @@ BucketManagerImpl::BucketManagerImpl(Application& app)
     , mBucketSnapMerge(app.getMetrics().NewTimer({"bucket", "snap", "merge"}))
     , mSharedBucketsSize(
           app.getMetrics().NewCounter({"bucket", "memory", "shared"}))
-    , mDeleteEntireBucketDirInDtor(app.getConfig().isInMemoryMode())
+    , mDeleteEntireBucketDirInDtor(
+          app.getConfig().isInMemoryMode() &&
+          !app.getConfig().MODE_STORES_HISTORY_LEDGERHEADERS)
```

Comment:
A comment would be nice here explaining why `!MODE_STORES_HISTORY_LEDGERHEADERS` relates to the deletion of entries in the bucketlist; or perhaps a helper method like `bool Config::isInMemoryModeWithoutMinimalDB() const { return isInMemoryMode() && !MODE_STORES_HISTORY_LEDGERHEADERS; }` or something.

Reply:
+1 on adding a helper function for clarity